procfs/src/process/mod.rs

```diff
@@ -144,15 +144,14 @@ impl FDInfo {
     ) -> ProcResult<Self> {
         let p = path.as_ref();
         let root = base.as_ref().join(p);
-        let file = wrap_io_error!(
-            root,
-            rustix::fs::openat(
-                dirfd,
-                p,
-                OFlags::NOFOLLOW | OFlags::PATH | OFlags::CLOEXEC,
-                Mode::empty()
-            )
-        )?;
+        // for 2.6.39 <= kernel < 3.6 fstat doesn't support O_PATH see https://github.com/eminence/procfs/issues/265
+        let flags = match *crate::KERNEL {
+            Ok(v) if v < KernelVersion::new(3, 6, 0) => OFlags::NOFOLLOW | OFlags::CLOEXEC,
+            Ok(v) if v >= KernelVersion::new(3, 6, 0) => OFlags::NOFOLLOW | OFlags::PATH | OFlags::CLOEXEC,
+            Ok(_) => unreachable!(),
```

Comment:
This is minor, but can we re-write the match to get rid of the unreachable statement?

```suggestion
            Ok(_) => OFlags::NOFOLLOW | OFlags::PATH | OFlags::CLOEXEC,
```